packages/buidler-core/src/internal/util/jsonrpc.ts

```diff
@@ -0,0 +1,93 @@
+import { BuidlerError } from "../core/errors";
+import { ERRORS } from "../core/errors-list";
+
+export interface JsonRpcRequest {
+  jsonrpc: string;
+  method: string;
+  params: any[];
+  id: number;
+}
+
+interface SuccessfulJsonRpcResponse {
+  jsonrpc: string;
+  id: number;
+  result: any;
+}
+
+export interface FailedJsonRpcResponse {
+  jsonrpc: string;
+  id?: number;
+  error: {
+    code: number;
+    message: string;
+    data?: any;
+  };
+}
+
+export type JsonRpcResponse = SuccessfulJsonRpcResponse | FailedJsonRpcResponse;
+
+export function parseJsonResponse(text: string): JsonRpcResponse {
+  try {
+    const json = JSON.parse(text);
+
+    if (!isValidJsonResponse(json)) {
+      throw new Error();
```

Comment:
Why not something like: `throw new InvalidJsonInputError(`Invalid JSON-RPC response: ${text}`);`?

Reply:
Oh, got it! 

Can we add a comment explaining it?

Reply:
Done.